### DISCUSSION TITLE
Whitelist a filter rule

### DISCUSSION THREAD

**Discussion by thany** [uBlockOrigin/uBlock-issues#3399] [General]
I found a website where codepen.io is blocked:
https://css-tricks.com/slide-through-unlimited-dimensions-with-css-scroll-timelines/

This blockage is coming from EasyList, according to the log at least. I *do want* the rules from EasyList to be applied, except this one rule I do *not* want applied.

So how can I whitelist this rule, or generally how can I whitelist any rule I don't want applied?

**Comment by gorhill** [comment]
Seems to work fine for me. What's not working? If you want to report a filter issue, use the _chat_ icon in the popup panel.

**Reply by thany** [reply]
I don't know what's not working. I'm just asking how to 🤷🏻‍♂️

Also it might not be an issue with the list or rule itself. I just want it not to apply on *my* machine. I never said the rule should be disabled on everyone's machine.

**Comment by Yuki2718** [comment]
Nothing in EasyList blocks any of `codepen.io` request here.

**Reply by thany** [reply]
That just what the log *appears* to tell me. It doesn't matter what is blocked where. The question is how to whitelist something, regardless of what exactly it is.

Edit: let me add an excerpt of the log then:
![image](https://github.com/user-attachments/assets/9149a0ec-d8e1-4ee0-aba2-d2525896753f)

And what an embedded codepen looks like after "running" it:
![image](https://github.com/user-attachments/assets/94c1fe84-d14c-451a-a461-1ec2553dba0d)

**Reply by stephenhawk8054** [reply]
The only blocking is CSP reports. You can go to uBO's settings, untick  `Block CSP reports` and test again.

**Reply by gorhill** [reply]
If this is the logger output when you load that codepen page, then uBO is not your issue, nothing in there is preventing the page to render properly, and it renders properly with default filter lists:

![image](https://github.com/user-attachments/assets/8e039fcc-21ed-4bf3-b216-b9e9f2dfd4c8)

**Comment by krystian3w** [comment]
> [!CAUTION]
> If you mean that animations don't work in Firefox 115-131, then in Firefox Nightly 133 it might start with modifying preferences:
> 
> `layout.css.scroll-driven-animations.enabled`

> [!WARNING]  
> **In releases below 133 only the first animation with the dice works with forced enabled these preference.**


Then you wrongly assume that some filter is corrupting the frames. 

---
---

> [!NOTE]
> Using the word rule when using uBo may mislead you into thinking you are using "firewall" mode.

**Reply by thany** [reply]
No, codepen frames are just empty because they are partly being blocked. I didn't say anything about animations, did I?

Btw, what other than "rule" should I call it then? I'm not being mislead I don't think. But regardless, I just want one rule (or whatever it's called) not applied. I'm just asking how to whitelist something. So I don't understand why you come up with such a strange followup.

**Comment by gothic-bum** [comment]
> or generally how can I whitelist any rule I don't want applied?
  
You can create filter exceptions using the UI in the logger:
https://github.com/gorhill/uBlock/wiki/The-logger#creating-filters 
  
Or you can create filter exceptions manually (see the links to filter syntax documentation and also the `badfilter` option):  
https://github.com/gorhill/uBlock/wiki/Static-filter-syntax
